### ISSUE TITLE
[FEATURE REQUEST] different keyboards - different shortcuts

### ISSUE THREAD

**Issue by eugenedm** [folivoraAI/BetterTouchTool#513]
I have 2 keyboards(1 usb and 1 bluetooth)… and I want to assign my USB-wired keyboard to do different commands from my bluetooth… like if I press key “a” on USB keyboard it executes an apple script a and if I press key “p” it executes multiple keystrokes... open folders, starts programs...

*THE ISSUE / SUGGESTION:  SUPPORT FOR MULTIPLE KEYBOARDS - ADD WHICH KEYBOARD THE SHORTCUT IS COMING FROM

 I want to be able to type on my bluetooth keyboard and use it as a "normal" keyboard  and I want my USB wired keyboard to do different thing like running scripts, press complex 3-key combinations for me etc... on the same keys... likE "a"-"z", "1"-"0"  or "tab" etc.. without effecting my main keyboard..

 I don’t want ctrl-shift-command-A  keys… They are toooooo long... And I used all of the F1-F19.. I need more keys to assign... 

 I want to use 2 keyboards separately and I want to them be treated as two separate devices with separate events… but there’s no way to let the program do that…

At the moment I'm also using “ControllerMate” and it almost can do this… and you can choose different keyboards there, but when I type “a” or “p” on my USB keyboard it executes the shortcut BUT ALSO it types the letter “a” or “p”…. but I want to suppress the typing… I like pressing 1 button and it does it all…they are too long and hard to remember…

so to summarize:

I want to:
#1. choose which keyboard to assign to
#2. assign a shortcut (on that specific keyboard)
#3. execute script

#2 , #3 is already supported in your software… which is cool!

but #1 is missing…

Thanks a lot… I hope this can be done…

#### Description of your bug/question/feature request...:

**What kind of Mac are you using?**
OSX SIERRA 27"iMac 2013

**Which input device (e.g. MacBook Trackad, Magic Mouse, Magic Trackpad Touch Bar) is affected by this?**
Magic Keyboard + USB Wired Keyboard 

**Which versions of macOS and BetterTouchTool are you running?**
2.071

**Other information:** (e.g. stacktraces, related issues, screenshots,suggestions how to fix, etc)
Suggestion: add which keyboard the shortcut is coming from...

**Comment by fifafu** [comment]
I like the idea, but it's not possible using the current BTT shortcuts implementation. I'll keep this open, but can't promise to implement it anytime soon as it will require a pretty big rewrite of some core components.

**Comment by eugenedm** [comment]
Thank you! I hope maybe in the future this could be possible... I think this feature could double the speed of work...... :)

**Comment by tpj2018** [comment]
For now just create a hyperkey mapped to capslock with karabiner and you’re done. its like pressing shift A.  
You could also use karabiner to map all the keys of the 2nd keyboard to commands you dont use and then use those to map to BTT.  I use a combination of BTT, karabiner and spacelauncher and very happy with it. I do agree having this feature would be great in BTT
